### PR TITLE
[3.6] Add precondition for distribute shards like prototype

### PR DESCRIPTION
**Pull request by goedderz**
Backport of #11901

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10875/

**Comment by KVS85** [comment]
Tests blue.

File: arangod/Agency/PathComponent.cpp [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 #include "PathComponent.h"
 
+#include <iostream>
```
Comment (neunhoef) [inline arangod/Agency/PathComponent.cpp:25]: Is this intentional? I think it could be removed.

File: arangod/Agency/Supervision.cpp [hunk 2]
```diff
@@ -34,10 +34,10 @@
 #include "Agency/JobContext.h"
 #include "Agency/RemoveFollower.h"
 #include "Agency/Store.h"
+#include "AgencyPaths.h"
```
Comment (neunhoef) [inline arangod/Agency/Supervision.cpp:37]: Why is this not `Agency/AgencyPaths.h`?

File: arangod/Agency/PathComponent.cpp (hunk 1 above)
Comment (goedderz) [reply arangod/Agency/PathComponent.cpp:25]: Right, that's a backport relic.

File: arangod/Agency/PathComponent.cpp [hunk 3]
```diff
@@ -22,6 +22,8 @@
...
+
```
Comment (goedderz) [inline arangod/Agency/PathComponent.cpp:26]:
```suggestion
```

File: arangod/Agency/Supervision.cpp (hunk 2 above)
Comment (goedderz) [inline arangod/Agency/Supervision.cpp:37]:
```suggestion
#include "Agency/AgencyPaths.h"
```

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10956/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**